### ISSUE TITLE
[Properties for shopaholic] Issue with values being updated incorrectly.

### ISSUE THREAD

**Issue by mrelevance** [oc-shopaholic/oc-shopaholic-plugin#222]
I have offers that contain custom properties called "size" and "diameter".
Size is a text value input - (e.g. 12" x .125 x 1" - 20mm)
Diameter - The diameter of the product in inches.

However, if the value of the size property is set to, let's say 4", and I set the diameter to 4. the value will be changed for the diameter value and will no longer be just a number.

To help better clarify, If I set one product offer with size = 12" x .125 and diameter = 4. The diameter value will be `4`. But then If I create a new offer with size = `4"` and diameter = `4`, the actual value in the database will be changed to `4"` because of how it is saving every value in the DB with a slug.

Since, it is using a slug for lookup or association (I haven't dug that far into it to determine what's actually happening), the Diameter value can sometimes be a string or an actual number depending on what has recently been updated/entered into either custom property field.

What should happen, is it I enter 4" into the size field, that shouldn't affect the diameter field value if they are the same, with the only difference being the `"`.

I'm not even sure if you will even be able to address this issue without a major breaking version based on your DB layout.

**Comment by mrelevance** [comment]
The table I'm referencing is `lovata_properties_shopaholic_values`

**Comment by kharanenka** [comment]
Hi! Sorry for the long reply.

Can you use measure for property values 4"?
For example: property value = 4, measure = "

**Comment by mrelevance** [comment]
Unfortunately no, because the size field is just a string value which can contain anything like `4"`, `12" x .125`, etc.

**Comment by kharanenka** [comment]
I need to think about your problem.

**Comment by kharanenka** [comment]
Do you use PropertyValue->slug field?
You can try to switch settings http://joxi.ru/E2p6zKLi7g6ber
This will save property value wichout changing.

**Comment by mrelevance** [comment]
I enabled the option to disable the Str::slug method, and regenerated all of the values slugs, but now when I update certain trade offers it tells me the slug is already taken and seems to be related to custom property values.

Can you think of an easy way to be able to solve this issue?


Thanks

**Comment by mrelevance** [comment]
After playing around with the custom property field it looks like it is related to it being a number input field and the way the database results are returned when supplied a literal number instead of a string.

It looks like when a literal number is supplied the SQL generated is as follows:
```sql
select * from `lovata_properties_shopaholic_values` where `value` = 7  
```
But if I set the input type to text
```sql
select * from `lovata_properties_shopaholic_values` where `value` = '7' 
```
Not sure if this is just specific to my database handling the SQL differently that it should be, but I am using MariaDB 10.2 if that helps.

**Comment by kharanenka** [comment]
Hi!

1. I checked the saving of property values if I changed the settings. Everything works correctly. How did you regenerate property values? It looks like you created duplicate property values in the database when you regenerated the values.
2. How do you pass property values ​​to a filter method? Are you getting a product filtering error?

**Comment by mrelevance** [comment]
1. I had looped through every property value using the save() method which forced it to regenerate the slug.
2. I am not getting any filtering errors.

Ever since I switched the field in the backend to be text from number, everything seems to be working correctly. Seems to be a DB SQL problem when using the number field as it then converts the input to a number in the SQL statement which causes issues because the DB slug column is text.

Going to close this since it's working good enough for me now.
